store-pg/src/main/scala/com/socrata/pg/store/PGSecondary.scala

```diff
@@ -133,23 +125,69 @@ class PGSecondary(val config: Config) extends Secondary[SoQLType, SoQLValue] wit
     }
   }
 
-  // Currently there are zero-or-more snapshots, which are what you get when you publish a working copy when there is
-  // an existing published working copy of that same dataset.
-  // To NO-OP this API, return an empty set.
-  @deprecated("Not supporting snapshots beyond bare minimum required to function", since = "forever")
-  def snapshots(datasetInternalName: String, cookie: Secondary.Cookie): Set[Long] = {
-    // if we a publish through version(); a snapshot "could" be created
-    logger.debug(s"snapshots '${datasetInternalName}' (cookie: ${cookie}})")
-    Set()
-  }
-
   // Is only ever called as part of a resync.
-  def dropCopy(datasetInternalName: String, copyNumber: Long, cookie: Secondary.Cookie): Secondary.Cookie = {
-    // TODO: dropCopy
-    logger.warn(s"TODO: dropCopy '${datasetInternalName}' (cookie: ${cookie}})")
+  def dropCopy(datasetInfo: DatasetInfo,
+               secondaryCopyInfo: SecondaryCopyInfo,
+               cookie: Secondary.Cookie, isLatestCopy: Boolean): Secondary.Cookie = {
+    logger.info("droping copy (datasetInfo: {}, secondaryCopyInfo: {}, cookie: {})",
+      datasetInfo, secondaryCopyInfo, cookie)
+    withPgu(dsInfo, Some(datasetInfo)) { pgu =>
+      doDropCopy(pgu, datasetInfo, secondaryCopyInfo, isLatestCopy)
+      pgu.commit()
+    }
     cookie
   }
 
+  def doDropCopy(pgu: PGSecondaryUniverse[SoQLType, SoQLValue], datasetInfo: DatasetInfo,
```

Comment:
But, it's not really an event. It's like resync. and the doResync is in this file. Keeping it here seems more consistent to me?